src/main/java/org/testobject/appium/testng/TestObjectTestNGTestProvider.java

```diff
@@ -0,0 +1,55 @@
+package org.testobject.appium.testng;
+
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.URL;
+
+public class TestObjectTestNGTestProvider {
+
+	private RemoteWebDriver remoteWebDriver;
+
+	private URL endpoint;
+
+	private boolean isLocalTest;
+
+	public static TestObjectTestNGTestProvider newInstance() {
+		return new TestObjectTestNGTestProvider();
+	}
+
+	private TestObjectTestNGTestProvider() {
+	}
+
+	public RemoteWebDriver getRemoteWebDriver() {
+		return remoteWebDriver;
+	}
+
+	public AppiumDriver getAppiumDriver() {
+		if (remoteWebDriver != null) {
```

Comment:
question is why the null check? you can cast null to everything so a simple 
return (AppiumDriver) varName;
would do it.